pennylane/pulse/parametrized_evolution.py

```diff
@@ -142,6 +156,11 @@ def circuit(params):
         operators are executed simultaneously, but rather that both evaluate their respective
         scalar-valued functions using the same time window. See Usage Details.
 
+    .. note::
+
+        Using ``return_intermediate`` in a quantum circuit leads to broadcasted execution,
+        which can lead to unintended additional computational cost.
```

Comment:
```suggestion
        ``return_intermediate=True`` should not be used in a quantum circuit as it leads to broadcasted execution,
        which can lead to unnecessary additional computational cost.
```
correct me if I'm wrong and there are situations where one would want to use that in a qnode/circuit

Reply:
We will use it ourselves for the shift rule. And users might do so as well: They may want to know how the expectation value looks after half of the evolution time or so? :)

Reply:
For parameter shift though, right? You would never do
```python
def qnode(params):
    qml.pulse.ParametrizedEvolution(H, params, t, return_intermediate=True)
    return qml.expval(qml.PauliZ(0))
```
directly inside a qnode right?

Reply:
Why not? You could do this and basically get the expectation values for all intermediate times, allowing you to observe the time evolution of the expectation value :)

Reply:
I didnt think this was possible with the current execution pipeline, but if that works that would of course be super handy!

Reply:
It is not possible with this PR alone, because the device uses the state vector evolution, but
- it's easy to make the device switch back to using the matrix, enabling this feature, and
- we may also consider making the state vector evolution compatible with this (by making it return intermediate state vectors as well). This would work for `complementary=False` but not for `complementary=True`

Reply:
nice! bite size chunks ftw! 🥳